tests/unit/Hooks/ImageHooksTest.php

```diff
@@ -99,19 +102,18 @@ public function testPostCreate() {
 		$this->logger->log('debug', 'Test Image Height: '.$imageHeight, array('app' => 'MediaMetadata'));
 		$this->logger->log('debug', 'Test Image Width: '.$imageWidth, array('app' => 'MediaMetadata'));
 
-		$this->imageDimension->expects($this->once())
-			->method('setImageId')
-			->with($jpgFile->getId());
-		$this->imageDimension->expects($this->once())
-			->method('setImageHeight')
-			->with($imageHeight);
-		$this->imageDimension->expects($this->once())
-			->method('setImageWidth')
-			->with($imageWidth);
-
-		$this->mapper->expects($this->once())
-			->method('insert')
-			->with($this->imageDimension);
+		$this->db->expects($this->once())
+			->method('lastInsertId')
+			->with($this->equalTo('*PREFIX*table'));
+
+		$this->imageDimension->setId($jpgFile->getId());
+		$this->assertContains('id', $this->imageDimension->getUpdatedFields());
+		$this->imageDimension->setImageHeight($imageHeight);
+		$this->assertEquals(2, count($this->imageDimension->getUpdatedFields()));
```

Comment:
@oparoz Actually, right now I had written this only to check if it was working but I had planned to test the values once things start working. 
Also, regarding the TestEntity class, I thought I can test it with the ImageDimension class as it is also an extension of the entity class. Do I have to create a fake TestEntity class? In that case, how will we test the ImageDimension class?